tests/syntax/test_specifiers.py

```diff
@@ -263,50 +615,227 @@ def test_in_heading_distribution():
         'ra = RectangularRegion(0@0, 0, 2, 2)\n'
         'ra.orientation = VectorField("foo", lambda pt: 1)\n'
         'rb = PolylineRegion([0 @ 0, 1 @ 1])\n'
-        'ego = Object in Uniform(ra, rb)'
+        'ego = new Object in Uniform(ra, rb)'
     )
     hs = [sampleEgo(scenario).heading for i in range(60)]
     h2 = pytest.approx(-math.pi/4)
     assert all(h == 1 or h == h2 for h in hs)
     assert any(h == 1 for h in hs)
     assert any(h == h2 for h in hs)
 
+# Contained In
+def test_contained_in_2d():
+    scenario = compileScenic(
+        'region = RectangularRegion(0@0, 0, 2, 2)\n'
+        'ego = new Object contained in region'
+    )
+    for _ in range(30):
+        scene = sampleScene(scenario, maxIterations=1000)
+        pos = scene.egoObject.position
+        assert -0.5 <= pos.x <= 0.5
+        assert -0.5 <= pos.y <= 0.5
+        assert pos.z == 0
+
+def test_contained_in_3d():
+    scenario = compileScenic(
+        'region = BoxRegion(dimensions=(5,5,5))\n'
+        'ego = new Object contained in region'
+    )
+    for _ in range(10):
+        scene = sampleScene(scenario, maxIterations=1000)
+        pos = scene.egoObject.position
+        assert -4.5 <= pos.x <= 4.5
+        assert -4.5 <= pos.y <= 4.5
+        assert -4.5 <= pos.z <= 4.5
```

Comment:
Shouldn't these ranges be from `-2.5` to `2.5`?

I'd also shrink the box to 2x2x2 to increase the chance of catching the containment constraint failing to be enforced (the very high iteration limit you imposed is still plenty to ensure the failure rate is small).